homeassistant/components/luxor/light.py

```diff
@@ -0,0 +1,199 @@
+
+
+async def async_safe_fetch(method):
+    """Safely fetch data."""
+    try:
+        # Note: asyncio.TimeoutError and aiohttp.ClientError are already
+        # handled by the data update coordinator
+        async with async_timeout.timeout(10):
+            return await method()
+    except aio_exceptions.ClientConnectorError as err:
+        raise UpdateFailed(f"Connection error: {err}")
+    except LuxorError as err:
+        raise UpdateFailed(f"Error communicating with API: {err}")
+
+
+def luxor_brightness_to_hass(value):
+    """Convert luxor brightness 1..100 to hass format 0..255."""
+    return int((value * 255) // 100)
+
+
+def hass_to_luxor_brightness(value):
+    """Convert hass brightness 0..255 to luxor 1..100 scale."""
+    return int(round((value * 100) / 255))
+
+
+class LuxorGroupLightEntity(CoordinatorEntity, LightEntity):
+    """Representation of a Luxor light group as a homeassisant light entity."""
+
+    def __init__(self, coordinator, client, group_id):
+        """Initialize the light."""
+        super().__init__(coordinator)
+        self.client = client
+        self.group_id = group_id
+
+    @property
+    def supported_features(self):
+        """Flag supported features."""
+        return SUPPORT_BRIGHTNESS
+
+    @property
+    def brightness(self):
+        """Return the brightness of this light between 0..255."""
+        return luxor_brightness_to_hass(self._data["Inten"])
+
+    @property
+    def is_on(self):
+        """Return true if the light is on."""
+        return self._data["Inten"] > 0
+
+    async def async_turn_on(self, **kwargs):
+        """Turn the light on."""
+        if ATTR_BRIGHTNESS in kwargs:
+            intensity = hass_to_luxor_brightness(kwargs[ATTR_BRIGHTNESS])
+        else:
+            intensity = 100
+
+        await self.client.illuminate_group(self.group_id, intensity=intensity)
+        await self.coordinator.async_request_refresh()
+
+    async def async_turn_off(self, **kwargs):
+        """Turn the light off."""
+        await self.client.illuminate_group(self.group_id, intensity=0)
+        await self.coordinator.async_request_refresh()
+
+    @property
+    def name(self):
+        """Return the name of the Luxor group."""
+        return self._data["Name"]
+
+    @property
+    def _data(self):
+        return self.coordinator.data[self.group_id]
+
+
+class LuxorThemeLightEntity(CoordinatorEntity, LightEntity):
+    """Representation of a Luxor theme as a homeassisant light entity."""
+
+    def __init__(self, coordinator, client, theme_id):
+        """Initialize the light."""
+        super().__init__(coordinator)
+        self.client = client
+        self.theme_id = theme_id
+
+    @property
+    def is_on(self):
+        """Return true if the light is on."""
+        return self._data["OnOff"] > 0
+
+    async def async_turn_on(self, **kwargs):
+        """Turn the light on."""
+        await self.client.illuminate_theme(self.theme_id, on_off=1)
```

Comment:
What happens if an exception is raised by `illuminate_theme`?

Reply:
It will throw an exception, that I assume will end up in the log, and the change won't go through.  Is there a best practice in ha for how to handle this?